packages/itwin/grouping-mapping-widget/src/widget/components/hooks/useMappingsOperations.ts

```diff
@@ -30,22 +31,82 @@ const fetchMappings = async (
   }
 };
 
+const fetchExtractionStatus = async (
+  iModelId: string,
+  getAccessToken: GetAccessTokenFn,
+  extractionClient: IExtractionClient,
+  setIconStatus: React.Dispatch<React.SetStateAction<"negative" | "positive" | "warning">>,
+  setIconMessage: React.Dispatch<React.SetStateAction<string>>,
+  setExtractionMessageData: React.Dispatch<React.SetStateAction<ExtractionMessageData[]>>
+) => {
+  setIconStatus("warning");
+  setIconMessage("Extraction status pending.");
+  try {
+    const accessToken = await getAccessToken();
+    const extractions = await extractionClient.getExtractionHistory(accessToken, iModelId);
+    const jobId = extractions[0].jobId;
+    const status = await extractionClient.getExtractionStatus(accessToken, jobId);
+    if (status.containsIssues) {
+      setIconStatus("negative");
+      setIconMessage("Extraction contains issues. Click to view extraction logs.");
+      let logs = await extractionClient.getExtractionLogs(accessToken, jobId);
```

Comment:
Use `const`. Filter the logs in the same statement. If you need this for debugging you will always have the Network tab in dev tools.

Reply:
I have changed let to const.